resize.php

```diff
@@ -0,0 +1,71 @@
+<?php
+require(dirname(__FILE__).'/config/config.inc.php');
+$url = filter_input(INPUT_GET, 'url');
+
+preg_match('/^(img\/([^\/]*)\/(?:\d\/)*)(\d+)-([_a-z0-9]+)\.([a-z]+)$/', $url, $matches);
+$url_path = $matches[1];
+$img_path = $matches[2];
+$id_image = $matches[3];
+$size_name = $matches[4];
+$ext = $matches[5];
+switch ($img_path) {
+    case 'p':
+        $type = 'products';
+        break;
+    case 'c':
+        $type = 'categories';
+        break;
+    case 'm':
+        $type = 'manufacturers';
+        break;
+    case 'su':
+        $type = 'suppliers';
+        break;
+    case 'st':
+        $type = 'stores';
+        break;
+}
+if (!isset($type)) {
+    display('404');
+    die();
+}
+$image_types = ImageType::getImagesTypes($type, true);
+foreach ($image_types as $item) {
+    if ($item['name'] == $size_name) {
+        $image_type = $item;
+        break;
+    }
+}
+if (!isset($image_types)) {
+    display('404');
+    die();
+}
+$doc_root = $_SERVER['DOCUMENT_ROOT'];
```

Comment:
I'm quite sure this logic must already exist somewhere (in an area of PrestaShop related) and we should be able to reuse it to [keep a single source of truth](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)